compiler/src/iree/compiler/Codegen/LLVMGPU/test/prefetch_shared_memory.mlir

```diff
@@ -150,3 +155,95 @@ func.func @noprefetch_copyback(%arg0: memref<128xf32>, %arg1: memref<128xf32>) {
   return
 }
 // CHECK-NOT: gpu.barrier
+
+// -----
+
+// CHECK-LABEL: @prefetch_scf_if
+func.func @prefetch_scf_if(%arg0: memref<128xf32>, %cond : i1) {
+  %cst = arith.constant dense<0.000000e+00> : vector<1xf32>
+  %cst_0 = arith.constant 0.000000e+00 : f32
+  %c128 = arith.constant 128 : index
+  %c1 = arith.constant 1 : index
+  %c0 = arith.constant 0 : index
+  %alloc = memref.alloc() : memref<1xf32, #gpu.address_space<workgroup>>
+  %alloca = memref.alloca() : memref<1xf32, #gpu.address_space<private>>
+  %0 = scf.for %arg1 = %c0 to %c128 step %c1 iter_args(%arg2 = %cst) -> (vector<1xf32>) {
+    scf.if %cond {
+      %1 = vector.transfer_read %arg0[%arg1], %cst_0 : memref<128xf32>, vector<1xf32>
+      vector.transfer_write %1, %alloca[%c0] : vector<1xf32>, memref<1xf32, #gpu.address_space<private>>
+    }
+    scf.if %cond {
+    %1 = vector.transfer_read %alloca[%c0], %cst_0 : memref<1xf32, #gpu.address_space<private>>, vector<1xf32>
+    vector.transfer_write %1, %alloc[%c0] {in_bounds = [true]} : vector<1xf32>, memref<1xf32, #gpu.address_space<workgroup>>
+    }
```

Comment:
Could you break this into two individual test cases? 

BTW, you've covered cases of:
 1) read no write
 2) write no read
 3) read and write
 4) I see this case missing: no read and no write

Reply:
I was thinking that will be follow up PR wanted things to be simple here but always correct.

Reply:
I didnt split becuase this test excercises the case 1. and 2. logic and it would be a lot of same checks splitting the tests.
I added a test for 4. though.